### PR TITLE
refactor grid.py for clarity

### PR DESCRIPTION
**Pull request by Simontwice**
# Description
refactor grid so it is more clear
this is extracted from the now withdrawn multiple yamls PR

# Checklist
Answer each question with "yes", and provide an explanation if the answer isn't what is expected.

1. Is this PR focused on a single feature without the possibility of being divided into smaller PRs?
   - Answer: yes
2. Are all variables set with logical defaults that are backward compatible?
   - Answer: yes
3. Can you provide a link to a Neptune experiment verifying the functionality of this PR?
   - Answer: no point
4. Have you successfully executed the linter and tests?
   - Answer: yes
5. Are all functions concise and don't require splitting?
   - Answer: yes
6. Is there documentation for arguments and complex logic?
   - Answer: yes
7. Are there no temporary solutions in the code? Should a task be added to Trello to refine the code?
   - Answer: yes
8. Are all functions placed in appropriate files?
   - Answer: yes
9. Is the PR name meaningful and descriptive?
   - Answer: yes
10. Is PR description provided, or unnecessary?
    - Answer: yes
  
Other questions:

11. Have you not made changes to the requirements or anything related to the Singularity image? If you have, did you generate and upload a new image to the clusters?
    - Answer: y
12. Is the change significant enough to notify all individuals working with this repository?
    - Answer: n
13. Is there a need for a second reviewer?
    - Answer: n

### CODE REVIEW

File: lizrd/scripts/grid.py [hunk 1]
```diff
@@ -6,244 +6,192 @@
...
-        NODELIST = "--nodelist=" + NODELIST
-
-    grid = create_grid(PARAMS)
-    grid = multiply_grid(grid, RUNS_MULTIPLIER)
-    no_experiments = len(grid)
-    minutes_per_exp = timestr_to_minutes(TIME)
-
-    if len(grid) > 1 and runner == MachineBackend.LOCAL and not DRY_RUN:
+
+    config["params"]["path_to_config"] = sys.argv[1]
```
Comment (sebastianjaszczur) [inline lizrd/scripts/grid.py:44]: = path

**Review by sebastianjaszczur — COMMENTED** [review]
LGTM from my side, just please get somebody else to approve it as well.

File: lizrd/scripts/grid.py [hunk 2]
```diff
@@ -6,244 +6,192 @@
...
+    total_minutes = 0
+
+    pprint.pprint(config)
+    single_exp_training_args_grid = create_grid(config["params"])
+
+    setup_args = get_setup_args_with_defaults(config, CLUSTER_NAME)
+    single_exp_training_args_grid = multiply_grid(
+        single_exp_training_args_grid, setup_args["runs_multiplier"]
+    )
+    no_experiments = len(single_exp_training_args_grid)
```
Comment (crewtool) [inline lizrd/scripts/grid.py:60]: AFAIK mamy konwencję, że  używamy n_experiments
```suggestion
    n_experiments = len(single_exp_training_args_grid)
```

File: lizrd/scripts/grid.py [hunk 3]
```diff
@@ -6,244 +6,192 @@
...
-    total_minutes = no_experiments * minutes_per_exp
-    if not runner == MachineBackend.LOCAL:
-        if not INTERACTIVE_DEBUG:
+    if not CLUSTER_NAME == MachineBackend.LOCAL:
+        if not interactive_debug_session:
             user_input = input(
-                f"Will run {no_experiments} experiments, using up {total_minutes} minutes, i.e. around {round(total_minutes / 60)} hours"
-                f"\nExperiment settings: \n{RUNNER=} \n{TIME=} \n{N_GPUS=} \nContinue? [Y/n] "
+                f"Will run {total_no_experiments} experiments, using up {total_minutes} minutes, i.e. around {round(total_minutes / 60)} hours\n"
+                f"Continue? [Y/n]"
```
[the remark this answers is not in the document]
Comment (Simontwice) [reply lizrd/scripts/grid.py:78]: ta PRka jest gotowa na to, że w następnej będziemy używać wielu configów na raz, a tam każdy będzie miał własnego runnera, własny czas i liczbę GPU, więc printowanie tego nie ma już za dużo sensu.

File: lizrd/scripts/grid.py [hunk 4]
```diff
@@ -6,244 +6,192 @@
...
         if user_input.lower() not in ("", "y", "Y"):
             print("Aborting...")
             exit(1)
 
-    slurm_command = "srun" if INTERACTIVE_DEBUG else "sbatch"
-
-    if not (INTERACTIVE_DEBUG or runner == MachineBackend.LOCAL):
-        exp_name = next(iter(grid))["name"]
+    if not (interactive_debug_session or CLUSTER_NAME == MachineBackend.LOCAL):
+        exp_name = grid[0][0]["name"]
```
Comment (crewtool) [inline lizrd/scripts/grid.py:89]: Na pierwszy rzut oka nie wiadomo o co chodzi.
Doczytałem, że grid to lista (training_args, setup_args)

```suggestion
        first_exp_training_args, _ = grid[0]
        exp_name = first_exp_training_args["name"]
```

**Review by crewtool — APPROVED** [review]

**Pull request merged**